features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/SyslogReceiverCamelNettyImpl.java

```diff
@@ -153,9 +172,17 @@ public void process(Exchange exchange) throws Exception {
                             // we are listening on an InetAddress, it will always be of type InetAddressSocket
                             InetSocketAddress source = (InetSocketAddress)exchange.getIn().getHeader(NettyConstants.NETTY_REMOTE_ADDRESS); 
                             // Syslog Handler Implementation to recieve message from syslogport and pass it on to handler
+                            
+                            // Increment the packet counter
+                            packetMeter.mark();
+                            
+                            // Create a metric for the syslog packet size
+                            packetSizeHistogram.update(buffer.toByteBuffer().capacity());
```

Comment:
Please assign buffer.toByteBuffer() to a variable and then use it here and in the SyslogConnection() call. We shouldn't call toByteBuffer() twice since this might be slower.